hazelcast/src/main/java/com/hazelcast/hotrestart/InternalHotRestartService.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2008-2016, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.hotrestart;
+
+import com.hazelcast.config.HotRestartPersistenceConfig;
+import com.hazelcast.internal.management.dto.ClusterHotRestartStatusDTO;
+import com.hazelcast.nio.Address;
+
+import java.util.Set;
+
+/**
+ * Internal service for interacting with hot restart related functionalities (e.g. force and partial start)
+ */
+public interface InternalHotRestartService {
```

Comment:
What about `HotRestartServiceIntegrationService`? Since this is not related to implementing Hot Restart functionality, just to interact with actual Hot Restart impl.